### PR TITLE
Add docs for api for pie-player and pie-controller

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: using/api/pie-controller.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# PieController 
+
+## ES5 access 
+When you load the controller.js in a script tag, a var _pie_controller is created, to give you access to the controller.
```
Comment (edeustace) [inline using/api/pie-controller.md:4]: We shouldn't create a global instance of a controller, there may be multiple instances on a page. also can we not use `_pie_x`? I'd propose anything `pie` related lives under a global `pie` object aka `var myController = new pie.Controller(model, env);`

File: using/api/pie-controller.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Javascript Api
+ 
+### Methods
+ 
+#### setConfig(json)
+Set the configuration object for the pies
+
+ ```
+  _pie_controller.setConfig({
```
Comment (edeustace) [inline using/api/pie-controller.md:26]: i'd prefer that the model/config is set in the constructor of the class definition - not set.

File: using/api/pie-player.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+The controller itself has an api to set the config.json.  
+In es5 the controller.js creates a var _pie_controller, that can be used to access the controller.  
+ 
+ ```
+  let config = ... //load json 
+  _pie_controller.setConfig(config);  
+  player.controller = _pie_controller
+ ```
+ 
+#### completeResponse()
```
Comment (edeustace) [inline using/api/pie-player.md:43]: isn't this outside the scope of `pie`?

**Comment by evaneus** [comment]
I only noticed these comments now.

let's follow up on how we want to handle communication around this.

In merging these changes I wasn't blessing the changes as approved, I was just getting the docs structure in place. I think we should - while the spec is changing so rapidly and isn't locked - take overall API design discussions elsewhere.

**Pull request merged**